### ISSUE TITLE
Segfault using Connection#port

### ISSUE THREAD

**Issue by cbandy** [ged/ruby-pg#491]
With pg 1.4.4, library_version 130008:

```bash
# happy
$ ruby -r pg -e 'puts PG::Connection.new(user: "postgres")'
#<PG::Connection:0x0000786712e77ad8>

# happy
$ ruby -r pg -e 'puts PG::Connection.new(user: "postgres").port'
5432

# happy
$ ruby -r pg -e 'puts PG::Connection.new(user: "postgres", port: nil)'
#<PG::Connection:0x00007d046978fe20>

# sad
$ ruby -r pg -e 'PG::Connection.new(user: "postgres", port: nil).port'
-e:1: [BUG] Segmentation fault at 0x0000000000000000

# sad
$ ruby -r pg -e 'PG::Connection.new(user: "postgres", port: "").port'
-e:1: [BUG] Segmentation fault at 0x0000000000000000
```

**Comment by cbandy** [comment]
When `port` is empty in a connection string, libpq does not set it from the environment nor the compile-time default. PQport then returns null, and we segfault in atoi.

**Comment by larskanis** [comment]
The fix was merged in #492 and is released as part of pg-1.4.5.
